pkg/odo/cli/delete/component/component.go

```diff
@@ -122,31 +124,63 @@ func (o *ComponentOptions) Run(ctx context.Context) error {
 // deleteNamedComponent deletes a component given its name
 func (o *ComponentOptions) deleteNamedComponent(ctx context.Context) error {
 	var (
-		list []unstructured.Unstructured
-		err  error
+		appName = odocontext.GetApplication(ctx)
+
+		clusterResources []unstructured.Unstructured
+		podmanResources  []*corev1.Pod
+		err              error
 	)
+	log.Info("Searching resources to delete, please wait...")
 	if o.clientset.KubernetesClient != nil {
-		log.Info("Searching resources to delete, please wait...")
-		list, err = o.clientset.DeleteClient.ListClusterResourcesToDelete(ctx, o.name, o.namespace)
+		clusterResources, err = o.clientset.DeleteClient.ListClusterResourcesToDelete(ctx, o.name, o.namespace)
+		if err != nil {
+			return err
+		}
+	}
+
+	if o.clientset.PodmanClient != nil {
+		_, podmanResources, err = o.clientset.DeleteClient.ListPodmanResourcesToDelete(appName, o.name)
 		if err != nil {
 			return err
 		}
 	}
-	if len(list) == 0 {
+
+	if len(clusterResources) == 0 && len(podmanResources) == 0 {
 		log.Infof(messageWithPlatforms(
 			o.clientset.KubernetesClient != nil,
 			o.clientset.PodmanClient != nil,
 			o.name, o.namespace,
 		))
 		return nil
 	}
-	printDevfileComponents(o.name, o.namespace, list)
+	printDevfileComponents(o.name, o.namespace, clusterResources)
+
+	if len(podmanResources) > 0 {
+		log.Printf("The following pods and associated volumes will be deleted from podman:")
+		for _, pod := range podmanResources {
+			fmt.Printf("\t- %s\n", pod.GetName())
+		}
+	}
+
 	if o.forceFlag || ui.Proceed("Are you sure you want to delete these resources?") {
-		failed := o.clientset.DeleteClient.DeleteResources(list, o.waitFlag)
-		for _, fail := range failed {
-			log.Warningf("Failed to delete the %q resource: %s\n", fail.GetKind(), fail.GetName())
+		if len(clusterResources) > 0 {
+			failed := o.clientset.DeleteClient.DeleteResources(clusterResources, o.waitFlag)
+			for _, fail := range failed {
+				log.Warningf("Failed to delete the %q resource: %s\n", fail.GetKind(), fail.GetName())
+			}
+			log.Infof("The component %q is successfully deleted from namespace %q", o.name, o.namespace)
 		}
-		log.Infof("The component %q is successfully deleted from namespace %q", o.name, o.namespace)
+
+		if len(podmanResources) > 0 {
+			for _, pod := range podmanResources {
+				err = o.clientset.PodmanClient.CleanupPodResources(pod)
```

Comment:
Optional:
It takes a few seconds to cleanup the pod resources, I think adding a spinner here would be a good idea.